src/modules/welcome/checker/RequirementsChecker.cpp

```diff
@@ -337,8 +339,20 @@ RequirementsChecker::checkHasPower()
 bool
 RequirementsChecker::checkHasInternet()
 {
-    // default to true in the QNetworkAccessManager::UnknownAccessibility case
-    bool hasInternet = QNetworkAccessManager(this).networkAccessible() != QNetworkAccessManager::NotAccessible;
+    // doc from: https://doc.qt.io/qt-5/qnetworkconfigurationmanager.html#isOnline
+    bool hasInternet = false;
+
+    QNetworkConfigurationManager mgr;
+    QList<QNetworkConfiguration> activeConfigs = mgr.allConfigurations(QNetworkConfiguration::Active);
+
+    if (activeConfigs.count() > 0) {
+        Q_ASSERT(mgr.isOnline());
+        hasInternet = mgr.isOnline();
```

Comment:
I agree, after some discussion in IRC is clear that this code is not robust enough. I see only as a valid solution a "ping" request to a defined (configurable) server. What do you think?